designs/operator-monitoring-plugin.md

```diff
@@ -0,0 +1,225 @@
+---
+title: controller-observation-plugin
+authors:
+  - "@Kavinjsir"
+reviewers:
+  - "@varsha"
+  - "@camila"
+approvers:
+creation-date: 2022-06-06
+last-updated: 2022-06-06
+status: provisional
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# a-grafana-dashboard-plugin-to-observe-controller-metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+- 1. What does the ideal approach for the plugin to create the grafana manifests?
+- 2. Should we scaffold a dir with the config to install the grafana? See: https://grafana.com/docs/grafana/latest/installation/kubernetes/
+- 3. Where should the default directory be scaffold?
+  - grafana/
+  - grafana/dashboards
+  - grafana/config
+  - OR inside of the config dir another grafana dir with the k8s manifests to be applied on the cluster when the plugin is used
+- 4. How is the plugin backwards compatible for users who have prometheus scaffolded?
+  - Is it compatible only with specific versions of prometheus? Or do can we have an existing kb project and add this plugin to scaffold grafana manifests smoothly. Do we need any other changes to an old project structure.
+
+## Summary
+
+This EP will provide a way for the operator author to visualize the runtime metrics in order to observe the operator status.
+
+We currently have controller-runtime metrics exposed by default for all KB operators at an endpoint through Prometheus.
+
+In order to visualize those same metrics, every operator author has to individually put in the effort to create a dashboard.
+
+This plugin intends to reduce the effort of operator authors to create dashboards and instead provides with the manifests that will allow them to integrate their controller metrics with grafana easily.
+
+On top of the basic dashboard provided by us having default metrics, operator authors can also add custom metrics and visualize them in the format they are comfortable in.
+
+The functionality will be implemented by a new plugin that can generate the 'manifests' to be loaded on Grafana.
+
+In the context of this EP, the Grafana 'manifests' are mainly:
```

Comment:
I would add these parts on the technical details, I think the summary like https://github.com/kubernetes-sigs/kubebuilder/pull/2708/files#r892017769 shows its value as @varshaprasad96 suggested and is a good overview.